tests/integration_tests/functional/test_cpu_multiple.py

```diff
@@ -17,3 +19,23 @@ def test_all_vcpus_online(uvm_any):
         uvm_any.ssh.check_output(f"cat /sys/devices/system/cpu/online").stdout.strip()
         == f"0-{uvm_any.vcpus_count - 1}"
     )
+
+
+def test_all_vcpus_have_same_features(uvm_any):
+    """
+    Check all vCPUs have the same features inside guest.
+
+    This test ensures Firecracker or CPU templates don't configure CPU features
+    differently between vCPUs.
+
+    Note that whether the shown CPU features are expected or not should be
+    tested in (arch-specific) test_cpu_features_*.py only for vCPU 0. Thus, we
+    only test the equivalence of all CPUs in the same guest.
+    """
+    # Get a feature list of all CPUs
+    field = "flags" if global_props.cpu_architecture == "x86_64" else "Features"
+    features = uvm_any.ssh.check_output(
+        f"cat /proc/cpuinfo | grep {field}"
```

Comment:
This will match "vmx flags" in addition to "flags". Is it intended? If not, I think then we can simplify it

```sh
grep -P "^(flags|Features)" |sort |uniq
```

and then `assert len(stdout.splitlines()) == 1`

Reply:
Good catch!

Reply:
Hmm, I gave it some thought, and I think the option of using `uniq` isn't helpful since it won't show the value when it fails.